torchvision/models/regnet.py

```diff
@@ -0,0 +1,688 @@
+    block_type: Callable[..., nn.Module] = ResBottleneckBlock,
+) -> Tuple[nn.Sequential, int]:
+    current_width = stem_width
+
+    blocks = []
+    for i, (
+        width_out,
+        stride,
+        depth,
+        group_width,
+        bottleneck_multiplier,
+    ) in enumerate(params.get_expanded_params()):
+        blocks.append(
+            (
+                f"block{i+1}",
+                AnyStage(
+                    current_width,
+                    width_out,
+                    stride,
+                    depth,
+                    block_type,
+                    norm_layer,
+                    activation,
+                    group_width,
+                    bottleneck_multiplier,
+                    params.se_ratio,
+                    stage_index=i + 1,
+                ),
+            )
+        )
+
+        current_width = width_out
+    return (nn.Sequential(OrderedDict(blocks)), current_width)
+
+
+class Classifier(nn.Module):
+    def __init__(self, in_channels: int, num_classes: int = 1000) -> None:
+        super().__init__()
+        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
+        self.fc = nn.Linear(in_features=in_channels, out_features=num_classes)
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.avgpool(x)
+        x = x.flatten(start_dim=1)
+        x = self.fc(x)
+        return x
+
+
+class RegNet(nn.Module):
+    def __init__(
+        self,
+        stem: nn.Module,
+        blocks: nn.Module,
+        classifier: nn.Module,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__()
+        self.stem = stem
+        self.blocks = blocks
+        self.classifier = classifier
+
+        # Init weights and good to go
+        self.reset_parameters()
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.stem(x)
+        x = self.blocks(x)
+        x = self.classifier(x)
+        return x
+
+    def reset_parameters(self) -> None:
+        # Performs ResNet-style weight initialization
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                # Note that there is no bias due to BN
+                fan_out = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
+                nn.init.normal_(m.weight, mean=0.0, std=math.sqrt(2.0 / fan_out))
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.ones_(m.weight)
+                nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, mean=0.0, std=0.01)
+                nn.init.zeros_(m.bias)
+
+
+def _regnet(arch: str, block_params: BlockParams, pretrained: bool, progress: bool, **kwargs: Any) -> RegNet:
+    norm_layer = kwargs["norm_layer"] if "norm_layer" in kwargs else partial(nn.BatchNorm2d, eps=1e-05, momentum=0.1)
+    activation = kwargs["activation"] if "activation" in kwargs else nn.ReLU
+    num_classes = kwargs["num_classes"] if "num_classes" in kwargs else 1000
```

Comment:
maybe `kwargs.get("num_classes", 1000)` is better.

Reply:
slightly less redundant ? taste and colours but I would prefer that indeed

Reply:
nit: I would probably pass these as named parameters similar to [other builder methods](https://github.com/pytorch/vision/blob/main/torchvision/models/mobilenetv3.py#L188).